### PR TITLE
[TS] Part 13: Refactors, JavaDocs, And More Tests

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
Make everything consistent and document behaviour. Add more tests.

**Concerns (what feedback would you like?)**:
Is everything named well, etc.

(Note that there is a lot of renaming here so git thinks it's much bigger than it actually is)

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3180)

### CODE REVIEW

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepableTimestampsTest.java [hunk 1]
```diff
@@ -98,21 +110,21 @@ public void noNextTimestampWhenUnreadableTsInSmallerPartitionForConservativeOnly
     }
 
     @Test
-    public void canReadNextTimestampWhenSweepTimestampInSamePartitionAndGreater() {
-        long sweepTs = setSweepTimestampAndGet(maxTsForFinePartition(TS_FINE_PARTITION));
+    public void noNextTimestampWhenWhenSweepTimestampInSamePartitionAndLower() {
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepableTimestampsTest.java]: nit: `WhenWhen`

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+        assertSweepableCellsHasNoEntriesBeforeTimestamp(LOW_TS + 1);
+        assertSweepableCellsHasEntryForTimestamp(tsSecondPartitionFine);
+        assertSweepableCellsHasEntryForTimestamp(getSweepTsCons());
+
+        // last swept timestamp: largestBeforeSweepTs
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSweepableCellsHasNoEntriesBeforeTimestamp(LOW_TS + 1);
+        assertSweepableCellsHasNoEntriesBeforeTimestamp(tsSecondPartitionFine);
+        assertSweepableCellsHasEntryForTimestamp(getSweepTsCons());
+    }
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java]: Coverage looks solid. I'll see if I can come up with a few more cases, but this looks very nice 🌟

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/WriteInfoPartitionerTest.java [hunk 3]
```diff
@@ -87,6 +87,8 @@ public void getStrategyQueriesKvsOnlyOnceForEachTable() {
             partitioner.getStrategy(getWriteInfoWithFixedCellHash(NOTHING, i));
             partitioner.getStrategy(getWriteInfoWithFixedCellHash(CONSERVATIVE, i));
         }
+        verify(mockKvs, times(1)).getMetadataForTable(NOTHING);
+        verify(mockKvs, times(1)).getMetadataForTable(CONSERVATIVE);
         verify(mockKvs, times(2)).getMetadataForTable(any());
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/WriteInfoPartitionerTest.java]: probably cleaner to use `verifyNoMoreInteractions(mockKvs);`

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java [hunk 4]
```diff
@@ -122,7 +139,7 @@ private void scheduleBackgroundThreads() {
         }
```
[the remark this answers is not in the document]
Comment (gmaretic) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java]: 1. 🏅 
2. is due to multithreading concerns of sweeping same shard multiple times while skipping others.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueDeleter.java [hunk 5]
```diff
@@ -25,13 +25,22 @@
...
     private final KeyValueService kvs;
 
-    public KvsSweepDeleter(KeyValueService kvs) {
+    SweepQueueDeleter(KeyValueService kvs) {
         this.kvs = kvs;
     }
 
+    /**
+     * Executes targeted sweep, by inserting ranged tombstones corresponding to the given writes, using the sweep
+     * strategy determined by the sweeper.
```
[the remark this answers is not in the document]
Comment (gmaretic) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueDeleter.java]: I think now it's better

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueDeleter.java [hunk 6]
```diff
@@ -52,10 +52,8 @@ public void sweep(Collection<WriteInfo> writes, Sweeper sweeper) {
...
 
     private Map<TableReference, Map<Cell, Long>> writesPerTable(Collection<WriteInfo> writes, Sweeper sweeper) {
-        Map<TableReference, Map<Cell, Long>> result = new HashMap<>();
-        writes.forEach(write -> result
-                .computeIfAbsent(write.writeRef().tableRef(), ignore -> new HashMap<>())
-                .put(write.writeRef().cell(), write.timestampToDeleteAtExclusive(sweeper)));
-        return result;
+        return writes.stream().collect(Collectors.groupingBy(
+                WriteInfo::tableRef,
+                Collectors.toMap(WriteInfo::cell, write -> write.timestampToDeleteAtExclusive(sweeper))));
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueDeleter.java]: 👍

**Pull request merged**